src/utils/promise.ts

```diff
@@ -0,0 +1,16 @@
+export interface ComponentWithUnmountState {
+  isComponentUnmounted: boolean;
+}
+
+export function bindPromiseToUnmountState<T>(
+  component: ComponentWithUnmountState,
+  promise: Promise<T>
+): Promise<T> {
+  return promise.then((results: T) => {
+    if (component.isComponentUnmounted) {
+      throw new Error('Promise was rejected on unmounted component');
```

Comment:
You probably don't want to throw anything here? Nothing was rejected, it was resolved but you don't want to resolve it.
So I would probably do something like this:
```js
return new Promise(async (resolve) => {
  const result = await promise;
  if (!component.isComponentUnmounted) {
    resolve(result);
  }
});
```

Reply:
yes, but next listener (which in component) might do something with component state, which will throw error

Reply:
in case if a component is unmounted the promise in your example will remain unresolved forever that is a memory leak.

Reply:
I don't understand? Next listener (isn't the component unmounted)?
Which state?

Reply:
Example code?

Reply:
@fcognite, misunderstood variable name `isComponentUnmounted`, my assumption was wrong

Reply:
But if it throws then you can an exception when it is unmounted? Isn't that problematic?

Reply:
Yes, I get "Uncaught in promise" exception on unmounted component unless "await" was wrapped in try/catch block.

Reply:
Yes, that something we want to avoid... Therefore I thought not resolving would be the best, but I understand the problem with memory leak. Hmm hard one...

Reply:
Another solution is to take rxjs with cancelable observables, but it's a pretty big dependency.

Reply:
No thanks. Can we have a isUnmounted check before we call `setState` instead of promise-cancel stuff?

Reply:
yes) this is the easiest solution, will push soon.